### PR TITLE
[feature]: ignoreEventCondition option

### PR DESCRIPTION
**Pull request by alpinagyok**
Hey!

### Changes

- added `ignoreEventCondition` option

### Why

We need a bit finer control of when to ignore events. As an example, checking `event.target.closest()` by role/css

Wdyt? 😄

### CODE REVIEW

**Comment by JohannesKlauss** [comment]
Please add some tests for that feature and add its API to the API documentation. Also I am not too happy about the name of the option. Maybe something like `ignoreEventWhen` is a more precise naming?

**Comment by alpinagyok** [comment]
Agree with the name 👍, thanks for the comment. Will add tests and docs in a moment

**Comment by alpinagyok** [comment]
@JohannesKlauss, done 👌

File: documentation/docs/api/use-hotkeys.mdx [hunk 1]
```diff
@@ -221,6 +221,33 @@ this might desirable. We can enable the callback trigger for an input tag using
...
+
+```ts
+enableOnFormTags: string[] // default: undefined
+```
+
+Normally we do not want a hotkey being triggered while a user types something into an input field. In some cases however
+this might desirable. We can enable the callback trigger for an input tag using the following values:
+
+`INPUT`, `TEXTAREA`, `SELECT`
+
```
Comment (JohannesKlauss) [inline documentation/docs/api/use-hotkeys.mdx:234]: This text is now doubled

File: tests/useHotkeys.test.tsx [hunk 2]
```diff
@@ -491,6 +491,60 @@ test('should ignore case of form tags', async () => {
...
+  const { getByTestId } = render(<Component cb={callback} ignoreEventWhen={eventCondition} />)
+
+  await user.keyboard('A')
+
+  expect(callback).toHaveBeenCalledTimes(1)
+
+  await user.click(getByTestId('test-button'))
+  await user.keyboard('A')
+
+  expect(callback).toHaveBeenCalledTimes(1)
```
Comment (JohannesKlauss) [inline tests/useHotkeys.test.tsx:518]: I don't quite understand this. The behavior I expected would be that the callback never gets called. Why would it get called the first time and what is the button click doing?

**Review by JohannesKlauss — CHANGES_REQUESTED** [review]

File: documentation/docs/api/use-hotkeys.mdx (hunk 1 above)
Comment (alpinagyok) [reply documentation/docs/api/use-hotkeys.mdx]: fixed 👍

**Review by JohannesKlauss — APPROVED** [review]

**Comment by JohannesKlauss** [comment]
Thank you for your work!

**Comment by Isti115** [comment]
Sorry to bump this already merged PR, but just out of curiosity I'd like to ask what the difference between `filter` and `ignoreEventWhen` is apart from the predicate being negated? Having both of these options only for the sake of one being affected by `filterPreventDefault` and the other not seems to be a bit redundant to me at first, but there might be something that I`m overlooking.

Also, if `ignoreEventWhen` is necessary, it should probably also be documented in the _Options_ table of the top level README.

**Comment by alpinagyok** [comment]
@Isti115, I didn't know those existed 😄. but now that I look at it, there's no mention of `filter` in the code. maybe it's a leftover documentation? probably a question for the maintainer

**Comment by Isti115** [comment]
@alpinagyok Wow, sorry for not checking thoroughly enough, you're completely right, it seems to have been removed at some point during the move to version 4 as far as I can tell from the commit history. I'll submit a PR to remove it from the documentation as well.

**Pull request merged**